docs/coding-style.md

```diff
@@ -45,23 +47,47 @@ Variable name `sut` (System Under Test) is recommended in unit tests that really
 * Dot before an invocation `.Method()` is placed at the beginning of a line.
 * The comma separating arguments is placed at the end of a line.
 * Method declaration parameters should be on the same line. If S103 is violated, parameters should be placed each on a separate line; the first parameter should be on the same line with the declaration; the other parameters should be aligned with the first parameter.
-```
-public void MethodWithManyParameters(int firstParameter,
-                                     string secondParameter,
-                                     Function<int, string, string> complexParameter);
-```
+    ```
+    public void MethodWithManyParameters(int firstParameter,
+                                         string secondParameter,
+                                         Function<int, string, string> complexParameter);
+    ```
+* Chained invocations and member accesses violating S103 can have a chain of properties and property-like methods (e.g. `GetRoot()`) on the first line. Every other `.Invocation()` or `.Member` should be on a separate line, aligned with a left-most single indendation.
```

Comment:
property-like methods are traps, see my [comment](https://github.com/SonarSource/sonar-dotnet/pull/5295#discussion_r790546438) about `DescendantNodes()`

> For example, DescendantNodes() without a filter can lead to major perf hit (e.g. SONARSEC-1603 and associated https://github.com/SonarSource/sonar-security/pull/1783).

Treating them like property access can hide their possible impact...

Reply:
There's no universal easy way out of it. So the `should` word should provide a way to solve anything out of ordinary

Reply:
I was more worried about the 

> can have a chain of properties and property-like methods (e.g. `GetRoot()`) on the first line

Reply:
```suggestion
* Chained invocations and member accesses violating S103 can have a chain of properties and property-like methods (e.g. `GetRoot()`) on the first line. Every other `.Invocation()` or `.Member` should be on a separate line, aligned with a left-most single indendation.
  * When property-like methods are computationally intensive (e.g. `DescendantNodes()`), they should be on a separate line.
```

Reply:
I'd leave it implicit to keep the rule simple. Putting it on another line of course doesn't violate the rule and there might be other aspects except for performance (like long names).

Reply:
I would also skip the mention of property-like methods and consider them as invocations. In my opinion, when formatting code we should not care much about how the method is implemented. 

We should also consider that method implementation can change in time. If this happens should we re-format the code?